NDTensors/src/lib/TypeParameterAccessors/test/test_wrappers.jl

```diff
@@ -0,0 +1,112 @@
+@eval module $(gensym())
+using Test: @testset
+using LinearAlgebra:
+  Adjoint,
+  Diagonal,
+  Hermitian,
+  LowerTriangular,
+  Symmetric,
+  Transpose,
+  UnitLowerTriangular,
+  UnitUpperTriangular,
+  UpperTriangular
+using NDTensors.TypeParameterAccessors:
+  TypeParameter,
+  is_wrapped_array,
+  parenttype,
+  set_eltype,
+  set_ndims,
+  set_parenttype,
+  unwrap_array_type
+using StridedViews: StridedView
+include("utils/test_inferred.jl")
+@testset "TypeParameterAccessors wrapper types" begin
+  @testset "Array" begin
+    array = randn(2, 2)
+    array_type = typeof(array)
+    @test_inferred parenttype(Matrix) == Matrix
+    @test_inferred is_wrapped_array(array) == false
+    @test_inferred parenttype(array) == array_type
+    @test_inferred set_eltype(array, Float32) isa Matrix{Float32}
+    @test_inferred set_eltype(array, Float32) ≈ array
+    @test_inferred set_eltype(Array{<:Any,2}, Float64) == Matrix{Float64}
+    @test_inferred set_ndims(Array{Float64}, 2) == Matrix{Float64} wrapped = true
+    @test_inferred set_ndims(Array{Float64}, TypeParameter(2)) == Matrix{Float64}
+    @test_inferred unwrap_array_type(array_type) == array_type
+  end
+  @testset "Base AbstractArray wrappers" begin
+    array = randn(2, 2)
+    for wrapped_array in (
+      Base.ReshapedArray(array, (2, 2), ()),
+      SubArray(randn(2, 2), (:, :)),
+      PermutedDimsArray(randn(2, 2), (1, 2)),
+    )
+      wrapped_array_type = typeof(wrapped_array)
+      @test parenttype(wrapped_array) == Matrix{Float64}
+      @test is_wrapped_array(wrapped_array)
+      @test unwrap_array_type(wrapped_array_type) == Matrix{Float64}
+      # TODO: Test `set_eltype`.
```

Comment:
Reminder to add this test in potentially.

Reply:
I added these missing tests.